src/main/java/com/base64/d/StudyGuideChecker.java

```diff
@@ -8,127 +8,120 @@
-                String patternString2 = "(https://content.infinitecampus.com/sis.*?)\"";
+                // Regex meanings:
+                // . => means any character
+                // + => means 1 or more
+                // * => means 0 or more
+                // ? => means 0 or 1
+                // Everything in the () is pattern/matcher "group" syntax, then \" is looking for the next quote
+                Pattern contentLinkPattern = Pattern.compile("(https://content.infinitecampus.com/sis.*?)\"");
 
-                Pattern pattern2 = Pattern.compile(patternString2);
+                Matcher contentLinkMatcher = contentLinkPattern.matcher(decodedText);
 
-
-                Matcher matcher2 = pattern2.matcher(decodedText);
-                //System.out.println("This is the matcher " + matcher2);
-
-
-                while (matcher2.find()) {
+                while (contentLinkMatcher.find()) {
                     count++;
-                    //some study guide links had '\' added for unknown reason. this code checks for it and removes it before adding it to the studyGuide.
-
-                    String shorterStudyGuideLink = matcher2.group(1).replaceAll("\\\\+$", "");
-//                            .substring(0, matcher2.group(1).length() - 1);
-//                System.out.println(shorterStudyGuideLink);
-//                  links are good because regex replace all removes "\" if there.
-                    studyGuideLinks.add(shorterStudyGuideLink);
-
+                    // Some study guide links had '\' added for unknown reason.
+                    // This code checks for it and removes it before adding it to the studyGuide.
+                    String shorterStudyGuideLink = contentLinkMatcher.group(1).substring(0, contentLinkMatcher.group(1).length() - 1);
+
+                    if (contentLinkMatcher.group(1).substring(contentLinkMatcher.group(1).length() - 1).equals("\\")) {
+                        studyGuideLinks.add(shorterStudyGuideLink);
+                    } else {
+                        studyGuideLinks.add(contentLinkMatcher.group(1));
+                    }
                 }
 
                 System.out.println(studyGuideLinks.size() + " Unique Campus Community links found");
-                //remove all the false positives that are in JSON of study guide but not visible to front end user.
-                studyGuideLinks.removeAll(KnownFalsePositives.BLACK_LIST);
+                // Remove all the false positives that are in JSON of study guide but not visible to front end user.
+                studyGuideLinks.removeAll(KnownFalsePositives.WHITE_LIST);
                 System.out.println(studyGuideLinks);
                 totalLinksFound += studyGuideLinks.size();
 
             } catch (IOException e) {
                 e.printStackTrace();
             }
 
-
+            // Test all links in a Study Guide
```

Comment:
There are two big blocks of code in this method. And at first I was confused what they were doing, so I just added a comment before the second block here, and also the block above. This is the type of thing that could be extracted into separate methods, to improve readability. But this would also require restructuring the code, because it currently is gathering all of the links in a study guide (first block), and then processing all links for the study guide (second block). It's possible that restructuring might be to check each link after it is retrieved (and just keep track of links that have already been checked, to prevent checking duplicates).